eva-accession-deprecate/src/main/java/uk/ac/ebi/eva/accession/deprecate/configuration/DeprecableClusteredVariantsReaderConfiguration.java

```diff
@@ -36,7 +37,21 @@
     @StepScope
     DeprecableClusteredVariantsReader deprecableClusteredVariantsReader(MongoClient mongoClient,
                                                                         MongoProperties mongoProperties,
-                                                                        MongoTemplate mongoTemplate) {
-        return new DeprecableClusteredVariantsReader(mongoClient, mongoProperties.getDatabase(), mongoTemplate);
+                                                                        MongoTemplate mongoTemplate,
+                                                                        InputParameters parameters) {
+        boolean assembliesProvided =
+                parameters.getAssemblyAccession() != null && !parameters.getAssemblyAccession().isEmpty();
+
+        if (parameters.getDeprecateAll() == assembliesProvided) {
```

Comment:
Why not make deprecateAll the default and only use assemblies when provided?

Reply:
that was the initial design, but I screwed the mongodev collections and almost screw production because of this specific detail. you see, when you are filling the properties files, you can misspell a parameter, specially when it has a similar but different name:

assemblyAccession vs assemblyAccesssions. 

Did you spot the typo? if you didn't, and you provided a list of assemblies to the second one, the list would have been ignored and you would have run the deprecation for all species, which we can't do and won't be able to do until goat and sheep are completely loaded.

I made deprecateAll=false the default because that's the default thing we are going to do in the next weeks.